### ISSUE TITLE
Docker run not working as expected

### ISSUE THREAD

**Issue by unidoct** [doitintl/secrets-init#43]
I have a Dockerfile which consists of:
```
ENTRYPOINT [ "/secrets-init", "--provider=google"]
CMD ["php-fpm"]
```
If I am using my Dockerfile in relation with `docker compose` it works as expected and variables are being replaced.
Commands like `docker compose run test sh -c "echo \$TEST "` show the replaced variable.
But when I am using the same image and I use `docker run --env-file=.env ... test sh -c "echo \$TEST" `, it doesn't show the replaced variable. I'd expect them to behave the same, but there must be a difference regarding the environment. Authentication with Google is working for both commands. Can't I use `/secrets-init` with `docker run ...`?

**Comment by razvan-agape** [comment]
Hey, @unidoct, [this](https://github.com/doitintl/secrets-init/pull/44) pr might do what you need. You can try it in your scenario/use-case to see if it works. It would be nice to have some feedback from someone else before merging it. Let me know how it goes.

**Comment by alexei-led** [comment]
Fixed with #44
